src/main/java/ladder/domain/Persons.java

```diff
@@ -2,33 +2,36 @@
 
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
-import java.util.Set;
 import java.util.stream.Collectors;
 
 public class Persons {
-    private final Set<Person> persons;
+    private final List<Person> persons;
 
-    private Persons(final Set<Person> persons) {
+    private Persons(final List<Person> persons) {
         this.persons = persons;
     }
 
     public static Persons from(String[] personNames) {
-        Set<Person> persons = Arrays.stream(personNames)
-                .map(name -> Person.from(name))
-                .collect(Collectors.toSet());
-        validate(personNames, persons);
-        return new Persons(persons);
+        validate(personNames.length, (int) Arrays.stream(personNames)
+                                                 .distinct()
+                                                 .count());
+        return Arrays.stream(personNames)
+                .map(Person::from)
+                .distinct()
+                .collect(Collectors.collectingAndThen(Collectors.toList(),
+                        Persons::new));
```

Comment:
가독성을 고려하여 이런식으로 구현하는건 어떠세요? `checkSize` 는 [guava checkNotNull](https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkNotNull(T)) 에서 사용하는 패턴이에요.
```suggestion
        return Arrays.stream(personNames)
                .map(Person::from)
                .distinct()
                .collect(collectingAndThen(
                        toList(),
                        persons -> new Persons(checkSize(persons, personNames.length))));
    }

    private static List<Person> checkSize(List<Person> persons, int expectedSize) {
        validate(expectedSize, persons.size());
        return persons;
    }
```
응용해서 checkSize 를 Util 로도 만들어 볼 수 있겠네요. (Util 로 가는경우, `Person` 타입이나 `persons` 간은 도메인 영역 용어를 사용하지 않는 부분도 주의깊게 보세요 😀 )
```java
    private static <T> List<T> checkSize(List<T> list, int expectedSize) {
        if (list.size() == expectedSize) {
            return list;
        }
        throw new IllegalArgumentException(
                String.format("리스트의 길이가 기대한 길이가 아닙니다. list size: %d, expected: %d", list.size(), expectedSize));
    }
```